File: backend/app/opponents/personalities.py
from __future__ import annotations

from app.opponents.schemas import PersonalityDefinitionResponse

PERSONALITIES: list[PersonalityDefinitionResponse] = [
    PersonalityDefinitionResponse(
        id="steady",
        name="Steady",
        accuracy_rate=0.80,
        base_response_time_ms=3500,
        response_time_variance_ms=175,
        speed_profile="uniform",
        tier_offset=0,
    ),
    PersonalityDefinitionResponse(
        id="speedster",
        name="Speedster",
        accuracy_rate=0.70,
        base_response_time_ms=3500,
        response_time_variance_ms=350,
        speed_profile="front_loaded",
        tier_offset=1,
    ),
    PersonalityDefinitionResponse(
        id="slow_starter",
        name="Slow Starter",
        accuracy_rate=0.75,
        base_response_time_ms=3500,
        response_time_variance_ms=280,
        speed_profile="back_loaded",
        tier_offset=0,
    ),
    PersonalityDefinitionResponse(
        id="unpredictable",
        name="Unpredictable",
        accuracy_rate=0.65,
        base_response_time_ms=3500,
        response_time_variance_ms=875,
        speed_profile="random",
        tier_offset=0,
    ),
    PersonalityDefinitionResponse(
        id="balanced",
        name="Balanced",
        accuracy_rate=0.78,
        base_response_time_ms=3500,
        response_time_variance_ms=245,
        speed_profile="uniform",
        tier_offset=0,
    ),
]

File: frontend/src/engine/race/personalities.ts
import type { AiPersonality } from './types';

export const STEADY: AiPersonality = {
  id: 'steady',
  name: 'Steady',
  accuracyRate: 0.80,
  baseResponseTimeMs: 3500,
  responseTimeVarianceMs: 175,
  speedProfile: 'uniform',
  tierOffset: 0,
};

export const SPEEDSTER: AiPersonality = {
  id: 'speedster',
  name: 'Speedster',
  accuracyRate: 0.70,
  baseResponseTimeMs: 3500,
  responseTimeVarianceMs: 350,
  speedProfile: 'front_loaded',
  tierOffset: 1,
};

export const SLOW_STARTER: AiPersonality = {
  id: 'slow_starter',
  name: 'Slow Starter',
  accuracyRate: 0.75,
  baseResponseTimeMs: 3500,
  responseTimeVarianceMs: 280,
  speedProfile: 'back_loaded',
  tierOffset: 0,
};

export const UNPREDICTABLE: AiPersonality = {
  id: 'unpredictable',
  name: 'Unpredictable',
  accuracyRate: 0.65,
  baseResponseTimeMs: 3500,
  responseTimeVarianceMs: 875,
  speedProfile: 'random',
  tierOffset: 0,
};

export const BALANCED: AiPersonality = {
  id: 'balanced',
  name: 'Balanced',
  accuracyRate: 0.78,
  baseResponseTimeMs: 3500,
  responseTimeVarianceMs: 245,
  speedProfile: 'uniform',
  tierOffset: 0,
};

export const PERSONALITIES: AiPersonality[] = [STEADY, SPEEDSTER, SLOW_STARTER, UNPREDICTABLE, BALANCED];

export async function fetchPersonalities(): Promise<AiPersonality[]> {
  const resp = await fetch('/api/v1/opponents/personalities');
  if (!resp.ok) {
    throw new Error(`GET /api/v1/opponents/personalities failed: ${resp.status}`);
  }
  return resp.json() as Promise<AiPersonality[]>;
}
